lib/online_kitchen/labmanager4.rb

```diff
@@ -75,6 +75,25 @@ def equip_machine_start(name)
         end
       end
 
+      def equip_create_clean_snapshot(name)
+        config = OnlineKitchen.config.rest_config
+        machine_id = /.*\-([^\-]*)/.match(name)[1]
+        uri = URI("#{config[:service_endpoint]}machines/#{machine_id}/snapshots")
+        OnlineKitchen.logger.debug("creating clean snapshot")
+        req1 = Net::HTTP::Post.new(uri)
+        req1.content_type = 'application/json'
+        req1.body = '{"name":"clean"}'
+        req1.basic_auth config[:username], config[:password]
+        request_id = Net::HTTP.start(uri.hostname, uri.port, config.http_opts.symbolize_keys) do |http|
+          response = http.request(req1)
+          JSON.parse(response.body)['responses'][0]['request_id']
+        end
+
+        OnlineKitchen.logger.debug("waiting for clean snapshot to be created")
+        wait_for_machine_released(config, request_id)
```

Comment:
I would propose to rename the method `wait_for_machine_released` to `wait_for_request_completion`.